### PR TITLE
Feat: oseth adapter

### PR DESCRIPTION
**Pull request by rustboyar**

### CODE REVIEW

File: foundry.toml [hunk 1]
```diff
@@ -8,6 +8,7 @@ remappings = [
 ]
 ffi = true
 fs_permissions = [{ access = "write", path = "./reports" }]
+evm_version = 'shanghai'
```
Comment (rustboyar) [inline foundry.toml:11]: call to the OsTokenVaultController doesn't work without this

Comment (eboadom) [reply foundry.toml:11]: Is shanghai the latest? Because it should be using the latest to be fully sure of the system working in production

Comment (rustboyar) [reply foundry.toml:11]: will update to cancun
strange, that by default it doesn't choose the latest one for me

Comment (rustboyar) [reply foundry.toml]: removed the `evm_version` setting as the problem was related with forking for tests

File: tests/utils/GetExchangeRatesTest.t.sol [hunk 2]
```diff
@@ -24,13 +24,14 @@ import {IChainlinkAggregator} from 'cl-synchronicity-price-adapter/interfaces/IC
...
+import {IOsTokenVaultController} from '../../src/interfaces/IOsTokenVaultController.sol';
 
-import {CapAdaptersCodeEthereum} from '../../scripts/DeployEthereumWeEth.s.sol';
+import {CapAdaptersCodeEthereum} from '../../scripts/DeployEthereum.s.sol';
 import {CapAdaptersCodeArbitrum} from '../../scripts/DeployArbitrumWeEth.s.sol';
 
 contract ExchangeRatesEth is Test {
   function setUp() public {
-    vm.createSelectFork(vm.rpcUrl('mainnet'), 19515330); // 26th of March
+    vm.createSelectFork(vm.rpcUrl('mainnet'), 19722981); // 20th of April
```
Comment (kyzia551) [inline tests/utils/GetExchangeRatesTest.t.sol:34]:
```suggestion
    vm.createSelectFork(vm.rpcUrl('mainnet'), 19722981); // 24th of April
```

**Review by kyzia551 — APPROVED** [review]

**Pull request merged**